Feat/dedup notifications (#71)
topfreegames/pusher: README.md

* `PUSHER_SENTRY_URL` - Sentry Client Key (DSN);

> Pusher can detect and filter out duplicate messages to avoid sending the same notification multiple times to the same device. 

Reviewer: ```suggestion
Pusher can detect and filter out duplicate messages to avoid sending the same notification multiple times to the same device. 
Note: This feature is not yet active. Currently, deduplication only tracks metrics and does not prevent duplicate messages.
```

* `PUSHER_DEDUP_REDIS_HOST` - Redis host for deduplication service
* `PUSHER_DEDUP_REDIS_PORT` - Redis port for deduplication service
* `PUSHER_DEDUP_REDIS_PASSWORD` - Redis password for deduplication service
* `PUSHER_DEDUP_TLS_DISABLED` - Whether to enable TLS for Redis connection

> * `PUSHER_<GCM/APNS>_DEDUP_TTL` - How long to remember duplicate messages (e.g., "1h")

Reviewer: ```suggestion
* `PUSHER_<GCM/APNS>_DEDUP_TTL` - `time.Duration` to remember duplicate messages (e.g., "1h", "1m")
```

> * `PUSHER_DEDUP_GAMES_<GAMENAME>_PERCENTAGE` - Game-specific percentage to check
> 
> ### Percentage-based Sampling
> 
> The deduplication system supports percentage-based sampling to reduce Redis load:
> - `0%`: Completely disable deduplication for a game
> - `100%`: Check all messages for duplicates
> - `1-99%`: Check only that percentage of messages
> 
> Sampling is based on a hash of the device ID, ensuring the same device always gets the same sampling decision.

Reviewer: ```suggestion
* `PUSHER_DEDUP_GAMES_<GAMENAME>_PERCENTAGE` - Sets the deduplication sampling rate for a specific game. 0% disables sampling, 100% checks all messages. Sampling is deterministic per device.
```

* `PUSHER_SENTRY_URL` - Sentry Client Key (DSN);

## Message Deduplication

Pusher can detect and filter out duplicate messages to avoid sending the same notification multiple times to the same device. 

* `PUSHER_DEDUP_REDIS_HOST` - Redis host for deduplication service
* `PUSHER_DEDUP_REDIS_PORT` - Redis port for deduplication service
* `PUSHER_DEDUP_REDIS_PASSWORD` - Redis password for deduplication service

> * `PUSHER_DEDUP_TLS_DISABLED` - Whether to disable TLS for Redis connection

Reviewer 2: I believe a better name to this would be `PUSHER_DEDUP_USE_TLS` or `PUSHER_DEDUP_TLS_ENABLED`.

Author: Ok

> * `PUSHER_DEDUP_TIMEFRAME` - How long to remember duplicate messages (e.g., "1h")

Reviewer 2: I believe a better name to this would be `PUSHER_DEDUP_TTL`.

Author: Ok!
